db/osqlsqlthr.c

```diff
@@ -144,15 +144,15 @@ static inline int osql_should_restart(struct sqlclntstate *clnt, int rc)
 #define RESTART_SOCKSQL_KEEP_RQID(keep_rqid)                                   \
     restarted = 0;                                                             \
     if (osql_should_restart(clnt, rc)) {                                       \
-        restarted = 1;                                                         \
         rc = osql_sock_restart(clnt, gbl_survive_n_master_swings, keep_rqid);  \
         if (rc) {                                                              \
             logmsg(LOGMSG_ERROR,                                               \
                    "%s: failed to restart socksql session rc=%d\n", __func__,  \
                    rc);                                                        \
+        } else {                                                               \
+            restarted = 1;                                                     \
         }                                                                      \
-    }                                                                          \
-    if (rc) {                                                                  \
+    } else if (rc) {                                                           \
```

Comment:
if `osql_sock_restart` fails, do we want to coalesce rcode to `SQLITE_INTERNAL`?

Reply:
Good point. Will undo this change.